network/betweenness.go

```diff
@@ -87,3 +93,57 @@ func Betweenness(g graph.Graph) map[int]float64 {
 
 	return cb
 }
+
+// BetweennessWeighted returns the betweenness centrality for nodes in the weighted
+// graph g used to construct the given shortest paths.
+//
+//  C_B(v) = \sum_{s ≠ v ≠ t ∈ V} (\sigma_{st}(v) / \sigma_{st})
+//
+func BetweennessWeighted(g graph.CostGraph, p search.ShortestPaths) map[int]float64 {
+	nodes := g.NodeList()
+	cb := make(map[int]float64, len(nodes))
```

Comment:
They are in cb above because I was iterating over the nodes. There is no real reason other than that to have them stored except for that. It wastes a bit of space when you aren't iterating over all nodes as a map (you could iterate over returned nodes from the graph instead and save the space), but these algorithms are not cheap for large n, so time for calculation probably dominates.

I guess we could go either way (unless I'm forgetting something that prompted me to put that in above - which I have a suspicion I am - if we dropped it, I'd need to experiment to see the results).

What do you think?